lib/improver/spotdata/common_functions.py

```diff
@@ -289,23 +291,33 @@ def list_entry_from_index(list_in, index_in):
     return list(zip(*list_in)[index_in])
 
 
-def datetime_constraint(time_in):
+def datetime_constraint(time_in, time_max=None):
     """
     Constructs an iris equivalence constraint from a python datetime object.
 
     Args:
     -----
     time_in : datetime.datetime object
         The time to be used to build an iris constraint.
+    time_max : datetime.datetime object
+        Optional max time, which if provided leads to a range constraint
+        being returned up to < time_max.
 
     Returns: iris.Constraint
         An iris constraint to be used in extracting data at the given time from
         a cube.
 
     """
-    return Constraint(
-        time=PartialDateTime(time_in.year, time_in.month,
-                             time_in.day, time_in.hour))
+    if time_max is not None:
```

Comment:
If you are OK with me being picky, positive `if` logic is generally better for readability (`if time_max is None`... `else`...) and `else` is not required after a return statement